go/libraries/doltcore/sqle/sqlfmt/row_fmt.go

```diff
@@ -113,10 +113,76 @@ func SqlRowAsInsertStmt(ctx context.Context, r sql.Row, tableName string, tableS
 	}
 
 	b.WriteString(")")
+	b.WriteString(" VALUES ")
 
-	b.WriteString(" VALUES (")
-	seenOne = false
+	// Write single insert
+	str, err := SqlRowAsSingleInsert(ctx, r, tableSch)
+	if err != nil {
+		return "", err
+	}
+	b.WriteString(str)
+
+	b.WriteString(";")
+
+	return b.String(), nil
+}
+
+func SqlRowAsBatchInsertStmtStart(ctx context.Context, r sql.Row, tableName string, tableSch schema.Schema) (string, error) {
```

Comment:
Structure it like this instead:

`InsertStatementPrefix(ctx, tableName, tableSch)`: returns `INSERT INTO ... VALUES` without any value tuples
`SqlRowAsTupleString`: returns value tuples

Then in your writer method, you call `Prefix` at the start of every batch, then always call `AsTupleString`